x/interchainstaking/keeper/callbacks.go

```diff
@@ -451,18 +452,14 @@ func AccountBalanceCallback(k Keeper, ctx sdk.Context, args []byte, query icqtyp
 	}
 
 	if coin.IsNil() {
-		denom := ""
-
-		for i := 0; i < len(query.Request)-len(accAddr); i++ {
-			if bytes.Equal(query.Request[i:i+len(accAddr)], accAddr) {
-				denom = string(query.Request[i+len(accAddr):])
-				break
-			}
-		}
-		// if balance is nil, the response sent back is nil, so we don't receive the denom. Override that now.
-		if err := sdk.ValidateDenom(denom); err != nil {
-			return err
+		// if the balance returned is zero for a given denom, we just get a nil response.
+		// lookup the denom from the request so we can set a zero value coin for the correct denom.
+		idx := bytes.Index(query.Request, accAddr)
+		if idx == -1 {
+			return errors.New("AccountBalanceCallback: invalid request query")
 		}
+		denom := string(query.Request[idx:])
```

Comment:
How come we are missing the sdk.ValidateDenom check?

Reply:
My reading of #194 was that the `ValidateDenom` was considered superfluous here. Thoughts?

Reply:
I had re-added ValidateDenom